### PR TITLE
UI Support for Creating Metrics Alerts

### PR DESCRIPTION
**Pull request by sonamgupta21**
# Description
Summarize the change.

Fixes #<issue-number> (link all the GitHub issues this addresses)

# Testing
Describe how you tested this code. How can the reviewers reproduce your tests?

# Checklist:
Before marking your pull request as ready for review, complete the following.

- [ ] I have self-reviewed this PR.
- [ ] I have removed all print-debugging and commented-out code that should not be merged.
- [ ] I have added sufficient comments in my code, particularly in hard-to-understand areas.
- [ ] I have formatted the code, if applicable. For Go, I have run `goimports -w .`.

### CODE REVIEW

File: static/js/alert.js [hunk 1]
```diff
@@ -280,17 +296,41 @@ function resetAddAlertForm(){
...
+        datePickerHandler(metricsQueryParams.start, metricsQueryParams.end, metricsQueryParams.start);
+        if(functionsArray){
+            allFunctions = await getFunctions();
+            functionsArray = allFunctions.map(function(item) {
+                return item.fn;
+            });
+        }
+        for (const index in metricsQueryParams.queries) {
+            const query = metricsQueryParams.queries[index];
+            console.log(query.query);
```
Comment (sunitakawane) [inline static/js/alert.js:326]: remove log

File: static/alert.html [hunk 2]
```diff
@@ -106,9 +113,10 @@
                                     </button>
                                     <div class="dropdown-menu box-shadow" id="data-source-options">
                                         <li id="option-1" class="data-source-option active">Logs</li>
+                                        <li id="option-1" class="data-source-option">Metrics</li>
```
Comment (sunitakawane) [inline static/alert.html:116]: change id="option-2"

File: static/js/metrics-explorer.js [hunk 3]
```diff
@@ -27,8 +27,13 @@ let availableMetrics = [];
 let previousStartEpoch = null;
 let previousEndEpoch = null;
 let rawTimeSeriesData=[];
-let allFunctions=[];
+let allFunctions,functionsArray =[];
+var availableOptions = ["max by", "min by", "avg by", "sum by", "count by", "stddev by", "stdvar by", "group by"];
```
Comment (nkunal) [inline static/js/metrics-explorer.js:31]: nit: we should name this var a little differnetly , it is availableOptions for what ?

**Comment by sunitakawane** [comment]
Screenshot 2024-06-26 at 9 19 32 PM
 On bigger screen , cancel/save btns are overlapping alerts div
Also can the timepicker div moved to extreme right?

**Review by sunitakawane — DISMISSED** [review]

**Review by nkunal — APPROVED** [review]

**Pull request merged**